### DISCUSSION TITLE
Building an x86 extension module against an x86 Python on an ARM macOS machine

### DISCUSSION THREAD

**Discussion by hauntsaninja** [PyO3/maturin#1980] [Q&A]
Currently, `python -m pip install -e .` on a maturin project on an ARM macOS machine using an x86 Python will give me something like:
```
ImportError: dlopen(/Users/shantanu/dev/watchfiles/watchfiles/_rust_notify.cpython-311-darwin.so, 0x0002): tried: '/Users/shantanu/dev/watchfiles/watchfiles/_rust_notify.cpython-311-darwin.so' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/System/Volumes/Preboot/Cryptexes/OS/Users/shantanu/dev/watchfiles/watchfiles/_rust_notify.cpython-311-darwin.so' (no such file), '/Users/shantanu/dev/watchfiles/watchfiles/_rust_notify.cpython-311-darwin.so' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64'))
```
As far as I can tell, `setuptools-rust` correctly determines target, I think by: https://github.com/PyO3/setuptools-rust/blob/d2f3b3a1e336061604f4eb0fce8c46df6b24e9a9/setuptools_rust/build.py#L700

Am I missing a way to make this work? :-)

**Comment by hauntsaninja** [comment]
Okay, setting `CARGO_BUILD_TARGET=x86_64-apple-darwin` manually is a workaround

**Comment by messense** [comment]
We actually have a similar check in `maturin develop`

https://github.com/PyO3/maturin/blob/855f6d2cb1fb8fb43c2bb9e500ab0e5e84bd3140/src/develop.rs#L245-L246

I guess we didn't do the same in `maturin pep517 build-wheel` command.
